SQL Store.adoc

```diff
@@ -0,0 +1,262 @@
+= SQL Store
+
+As a application developer or data grid administrator, I want to preload a cache from a relational database accessed by JDBC - via configuration without writing boilerplate code.
+
+== Architecture
+
+* the store can be either read/write or read-only (in which case it behaves as a loader)
+* the store can work in either single-table mode or in query-mode (via user supplied queries)
+* the store can work with "primitive" keys/values or with complex types defined as protobuf schemas
+* a protobuf schema can either be generated dynamically from the resultset metadata or supplied by the user
+* the key can either be a single column or composite
+* the store is to be implemented as part of the existing JDBC store module so that it can reuse the general configuration of connection pooling/datasources and the dialect helpers
+
+== Modes
+
+=== Table mode
+
+This mode is the simplest, in that it just requires the user to supply a single table name.
+Through the database metadata, the store will determine:
+
+* the column names and types
+* which columns are part of the primary key
+
+The proposed XML configuration for this mode:
+
+[source,xml]
+----
+<local-cache name="books">
+   <persistence>
+      <table-jdbc-store>
+         <data-source jndi-url="..." />
+         <!-- the table element is required -->
+         <table name="books"/>
+         <!-- the schema element is optional. Defaults will be used otherwise -->
+         <schema name="Book" key-name="BookID" package="library"/>
```

Comment:
How do we specify a composite key here? Should it be `key-names="..."`?

Reply:
that is just the ProtoBuf message name. See further down for an example.

Reply:
Should it be called "message-name" then?